acmOpportunities/main.py

```diff
@@ -0,0 +1,381 @@
+
+    with instantiates_db_connection() as connection:
+        cursor = connection.cursor()
+
+        for job in job_data:
+            cursor.execute(
+                f"SELECT * FROM {table_name} WHERE _company = %(_company)s AND _title = %(_title)s AND _location = %(_location)s",
+                {
+                    "_company": job["_company"],
+                    "_title": job["_title"],
+                    "_location": job["_location"],
+                },
+            )
+            row = cursor.fetchone()
+
+            if row is None:
+                cursor.execute(
+                    f"INSERT INTO {table_name} (_company, _title, _location, _link, _processed) VALUES (%s, %s, %s, %s, %s)",
+                    (job["_company"], job["_title"], job["_location"], job["_link"], 0),
+                )
+        connection.commit()
+
+
+# ----------------- JOB DATA -----------------
+
+
+def rapid_response() -> List[object]:
+    """
+    This API call retrieves a formatted response object
+    and returns a List[object] as the result
+    """
+
+    url = os.getenv("RAPID_API_URL")
+
+    rapid_api_key = os.getenv("RAPID_API_KEY")
+    rapid_api_host = os.getenv("RAPID_API_HOST")
+
+    headers = {
+        "X-RapidAPI-Key": rapid_api_key,
+        "X-RapidAPI-Host": rapid_api_host,
+    }
+
+    rapid_jobs = []
+    response = requests.get(url, headers=headers).json()
+
+    command_line_value = extract_command_value()  # Extracts command-line value
+
+    for elem in response["hits"]:
+        time = elem["formatted_relative_time"]
+
+        numeric = re.search(r"\d+", time)
+        formatted_time_integer = int(numeric.group()) if numeric else 0
+
+        if len(rapid_jobs) <= 5 and int(command_line_value) >= formatted_time_integer:
+            job = {}
+            job["_company"] = elem["company_name"]
+            job["_title"] = elem["title"]
+            job["_location"] = elem["location"]
+            job["_link"] = f'https://www.indeed.com/viewjob?jk={elem["id"]}&locality=us'
+
+            rapid_jobs.append(job)
+
+    return rapid_jobs
+
+
+def linkedin_response() -> List[object]:
+    """Returns a List[object] which contains web scraped job content"""
+
+    url = os.getenv("LINKEDIN_URL")
+
+    response = requests.get(url)
+
+    content = response.text
+
+    parse_content = BeautifulSoup(content, "html.parser")
+
+    div_post = parse_content.find_all(
+        "div",
+        class_="base-card relative w-full hover:no-underline focus:no-underline base-card--link base-search-card base-search-card--link job-search-card",
+    )
+
+    date_div_post = parse_content.find_all(
+        "time",
+        class_="job-search-card__listdate",
+    )  # Recieves all the job posting dates
+
+    command_line_value = extract_command_value()  # Extracts command-line value
+
+    linked_in_jobs = []
+
+    for elem, date in zip(div_post, date_div_post):
+        """
```

Comment:
Docstrings are generally only intended to be used as ["the first statement in a module, function, class, or method definition"](https://peps.python.org/pep-0257/#what-is-a-docstring)

Reply:
Resolved: https://github.com/acmcsufoss/acmOpportunities/pull/1/commits/ab61bc4ad59cbcf21b8df00d42f1b8e9a9bbd854